consensus/broadcast/shardChainMessenger_test.go

```diff
@@ -451,6 +488,23 @@ func TestShardChainMessenger_BroadcastBlockDataLeaderNilMiniblocksShouldReturnNi
 	assert.Nil(t, err)
 }
 
+// This function should return ErrMockMarshalizer from core.CalculateHash
+func TestShardChainMessenger_BroadcastBlockDataLeaderNilMiniblocksShouldErrMockMarshalizer(t *testing.T) {
+	marshalizer := mock.MarshalizerMock{
+		Fail: true,
+	}
+
+	args := createDefaultShardChainArgs()
+	args.Marshalizer = marshalizer
+
+	scm, _ := broadcast.NewShardChainMessenger(args)
+
+	_, header, miniblocks, transactions := createDelayData("1")
+
+	err := scm.BroadcastBlockDataLeader(header, miniblocks, transactions, []byte("pk bytes"))
```

Comment:
Mismatch between test name and what is being tested.

The test name suggests nil miniblocks will be passed to BroadcastBlockDataLeader, but it is not clear in the test if miniblocks is nil or not. 

also in case miniblocks is nil, the function BroadcastBlockDataLeader returns nil immediately instead of throwing an error, just like for empty miniblocks.

Reply:
same mistake here, my intention was not to test the NilMiniBlocks but I forgot once more to delete that part from the function name when I copied it.